### PR TITLE
analysis: Move global imports to object-specific analysis and under versioning constraints

### PR DESCRIPTION
**Pull request by MarijnS95**
While adding some new classes to gstreamer-rs gir keeps emitting `use glib::translate::*;` for classes that do not have any members that could possibly use translation functionality, resulting in obvious "unused import" linter warnings. ~Simply adding this import only if there are members (functions, signals or properties) solves that problem for now.~ All these cases have been addressed by adding `with_defaults` import wrappers, and importing "global" imports under these so that they have the appropriate version constraints. In cases where no members exist (or they are commented out) this code does not even run at all, and the import is completely omitted (the original issue this PR was about).

### CODE REVIEW

**Comment by MarijnS95** [comment]
This isn't solving all of it though. In a different bringup there are classes (3 of them are shared classes, one is an object) with parameters that cannot be generated (`/*Ignored*/` and `/*Unimplemented*/`). Probably need to find a solution to those before merging as well..

**Comment by sdroege** [comment]
> Probably need to find a solution to those before merging as well.

Should be enough to have a separate bool for those that also checks in addition if the item should be commented out, or not?

**Comment by MarijnS95** [comment]
> Should be enough to have a separate bool for those that also checks in addition if the item should be commented out, or not?

@sdroege The total endeavour was more fun than that, but all my `"unused import"` errors in `gstreamer-rs` have been resolved now and there are quite a few nice fixes in it for `gtk-rs` as well! Not that those aren't hidden by `#[allow(unused_imports)]` though...

**Comment by sdroege** [comment]
>  [@MarijnS95](/MarijnS95) [MarijnS95](/MarijnS95) marked this pull request as draft [16 minutes ago](#event-4269040387)

What's left to be done to make it complete? Can you put a small todo list into the issue description?

**Comment by MarijnS95** [comment]
@sdroege Noticed that after moving the `glib::translate` import to `signals`/`properties`/`special_functions` it can and should simply be removed from `analysis/object`. Wanted to test and implement that theory before someone got to the merge button, hence marked it as draft :)

**Comment by MarijnS95** [comment]
Cool, the tests for gtk-rs and all clippy tests for gstreamer-rs (where `#[allow(unused_imports)]` has been removed) succeed: https://gitlab.freedesktop.org/MarijnS95/gstreamer-rs/-/pipelines/263964.

Note that somehow `gstreamer-base` was forgotten about, which still had a bunch of unconstrained imports in `enums.rs`. Should all be fixed now!

**Review by sdroege — APPROVED** [review]
Looks good to me, thanks!

@GuillaumeGomez ?

**Review by GuillaumeGomez — APPROVED** [review]
That's super cool! Thanks!

**Pull request merged**
